servlet/src/main/java/io/undertow/servlet/spec/PushBuilderImpl.java

```diff
@@ -112,15 +112,14 @@ public PushBuilderImpl(HttpServletRequestImpl servletRequest) {
             if(cookie.getValue().getMaxAge() != null && cookie.getValue().getMaxAge() <= 0) {
                 //remove cookie
                 List<String> existing = this.headers.getAll(HttpHeaderNames.COOKIE);
-                if(existing != null) {
-                    Iterator<String> it = existing.iterator();
-                    while (it.hasNext()) {
-                        String val = it.next();
-                        if(val.startsWith(cookie.getKey() + "=")) {
-                            it.remove();
-                        }
+                List<String> newValues = new ArrayList<>(existing.size());
```

Comment:
Are you sure `existing` cannot be `null`?

Reply:
According to the docs, it can never be `null`

Reply:
I also checked all the implementation and saw that they always return an empty list if the name does not exist

Reply:
The old code was more defensive.